lib/jsdom/living/events/MouseEvent-impl.js

```diff
@@ -5,8 +5,42 @@ const EventModifierMixinImpl = require("./EventModifierMixin-impl").implementati
 const UIEventImpl = require("./UIEvent-impl").implementation;
 
 const MouseEventInit = require("../generated/MouseEventInit");
+const { wrapperForImpl } = require("../generated/utils");
 
 class MouseEventImpl extends UIEventImpl {
+  get x() {
+    return this.clientX;
+  }
+  get y() {
+    return this.clientY;
+  }
+  get pageX() {
+    if (this._dispatchFlag) {
+      return 0;
+    }
```

Comment:
@jsnajdr when implementing this, my goal was to:
1. Follow the spec as closely as possible, and
2. Return something reasonable where we can't fully implement it, i.e. since jsdom doesn't implement layout/rendering, some things are unknowable so perhaps 0 is the least bad option

A (minor) risk with returning the current `clientX + scrollX` is that it could be inconsistent with browsers, since `scrollX/Y` could change between the time of event initialization and when `pageX`/`pageY` are read during dispatch (e.g. if explicitly set or `scrollTo` is called during an earlier click handler).

That said, it's probably still a better default than zero, and assuming `scrollX/Y` hasn't changed, it should be synonymous with `the position where the event occurred relative to the origin of the initial containing block`.

@domenic wdyt? Seems like a reasonable change to me, though perhaps it's still a good idea to keep the dispatch branch even if the return value is the same. That way it's a good placeholder/indicator for when layout support is added.

Reply:
I do have an [existing PR](https://github.com/jsdom/jsdom/pull/3496) that is closely related to this, and I think we can solve the minor risk outlined above by setting the new [`_initialContainingBlockRelativeX/Y`](https://github.com/jsdom/jsdom/pull/3496/files#diff-10b0a4b9fcc799389fd0a69fdce632ca6b9b82319f1ce74fa005913a9755a6f8R49-R50) to `clientX/Y + scrollX/Y`.

That way the values will be persisted at init time, and not risk changing if `scrollX/Y` changes later before/during dispatch.

Maybe I'll just split it out into its own PR

Reply:
#3514 should resolve this